test/src/main/java/org/apache/accumulo/test/IteratorEnvIT.java

```diff
@@ -71,121 +92,123 @@ public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoo
     @SuppressWarnings("deprecation")
     ServiceEnvironment serviceEnv = env.getServiceEnv();
     PluginEnvironment pluginEnv = env.getPluginEnv();
-    if (serviceEnv != pluginEnv) {
-      throw new RuntimeException("Test failed - assertSame(getServiceEnv(),getPluginEnv())");
-    }
+    assertEquals(serviceEnv, pluginEnv, "Test failed - assertSame(getServiceEnv(),getPluginEnv())");
 
     // verify property exists on the table config (deprecated and new),
     // with and without custom prefix, but not in the system config
     @SuppressWarnings("deprecation")
-    String accTableConf = env.getConfig().get("table.custom.iterator.env.test");
-    if (!"value1".equals(accTableConf)) {
-      throw new RuntimeException("Test failed - Expected table property not found in getConfig().");
-    }
+    String accTableConf = env.getConfig().get(CUSTOM_PROP_KEY);
+    assertEquals(CUSTOM_PROP_VAL, accTableConf,
+        "Test failed - Expected table property not found in getConfig().");
     var tableConf = pluginEnv.getConfiguration(env.getTableId());
-    if (!"value1".equals(tableConf.get("table.custom.iterator.env.test"))) {
-      throw new RuntimeException("Test failed - Expected table property not found in table conf.");
-    }
-    if (!"value1".equals(tableConf.getTableCustom("iterator.env.test"))) {
-      throw new RuntimeException("Test failed - Expected table property not found in table conf.");
-    }
+    assertEquals(CUSTOM_PROP_VAL, tableConf.get(CUSTOM_PROP_KEY),
+        "Test failed - Expected table property not found in table conf.");
+    assertEquals(CUSTOM_PROP_VAL, tableConf.getTableCustom(CUSTOM_PROP_KEY_SUFFIX),
+        "Test failed - Expected table property not found in table conf.");
     var systemConf = pluginEnv.getConfiguration();
-    if (systemConf.get("table.custom.iterator.env.test") != null) {
-      throw new RuntimeException("Test failed - Unexpected table property found in system conf.");
-    }
+    assertNull(systemConf.get(CUSTOM_PROP_KEY),
+        "Test failed - Unexpected table property found in system conf.");
 
     // check other environment settings
-    if (!scope.equals(env.getIteratorScope())) {
-      throw new RuntimeException("Test failed - Error getting iterator scope");
-    }
-    if (env.isSamplingEnabled()) {
-      throw new RuntimeException("Test failed - isSamplingEnabled returned true, expected false");
-    }
-    if (!expectedTableId.equals(env.getTableId())) {
-      throw new RuntimeException("Test failed - Error getting Table ID");
+    assertEquals(scope, env.getIteratorScope(), "Test failed - Error getting iterator scope");
+    assertFalse(env.isSamplingEnabled(),
+        "Test failed - isSamplingEnabled returned true, expected false");
+    assertEquals(expectedTableId, env.getTableId(), "Test failed - Error getting Table ID");
+    assertNull(env.getSamplerConfiguration());
+    assertThrows(Exception.class, env::cloneWithSamplingEnabled);
```

Comment:
should we configure sampling for the tables in this test somehow and verify the result of env.cloneWithSamplingEnabled? Or is this check sufficient for this test

Reply:
good question for @keith-turner

Reply:
I can look into generally improving test when I look at #5500.  I was just thinking of RFileScanner when  I opened that issue, but I can look at making it more general.